java/WordCountOptimized.java

```diff
@@ -6,41 +6,26 @@
 import java.util.Collections;
 import java.util.Comparator;
 import java.nio.charset.StandardCharsets;
-import java.nio.ByteBuffer;
-
-import gnu.trove.map.custom_hash.TObjectIntCustomHashMap;
-import gnu.trove.map.hash.TObjectIntHashMap;
-import gnu.trove.strategy.HashingStrategy;
-import gnu.trove.procedure.TObjectIntProcedure;
+import net.openhft.hashing.LongHashFunction;
 
 /** Highly optimized word count algorithm for java incorporating multiple optimizations
- *  Key tricks: 
- *    - Use Trove primitive collections (performance and memory savings, plus direct-iteration procedures)
- *    - Buffer I/O, either using Buffered I/O classes or by explicitly reading chunks to an array    
+ *  Key tricks:
+ *    - Buffer I/O, either using Buffered I/O classes or by explicitly reading chunks to an array
  *    - Process recurring tokens separately from singletons (big performance gain)    
  *    - Work directly with raw bytes (safe for UTF-8), to reduce memory and avoid text decoding cost
  *    - Sort strings using byte order (fudging the sort order slightly, faster than decoding to sort)
  *    - Use a radix byte sort to sort large lists of bytestrings (small performance gain)
+ *    - Faster hashing
  *  
  *  Authors:  
  *   - Sam Van Oort <samvanoort@gmail.com> (svanoort on GitHub)
  *   - Rick Hendricksen (xupwup on GitHub)
  *   - sgwerder on GitHub
+ *   - Roman Leventov (leventov on Github)
  */
 class WordCountOptimized {
 
-    private static class BytesHashingStrategy implements HashingStrategy<byte[]> {
-        @Override
-        public int computeHashCode(byte[] object) {
-            return Arrays.hashCode(object);
-        }
-
-        // Compare byte-byte byte, if they're equal aside from length, return last
-        @Override
-        public boolean equals(byte[] b1, byte[] b2) {
-            return b1 == b2 || Arrays.equals(b1, b2);
-        }
-    } 
+    public static final LongHashFunction XX_HASH = LongHashFunction.xx_r39();
```

Comment:
It is simply faster: https://github.com/OpenHFT/Zero-Allocation-Hashing#performance